### PR TITLE
[2.7] Enhance weighted agg helper

### PR DESCRIPTION
**Pull request by YuanTingHsieh**
Takes changes from https://github.com/NVIDIA/NVFlare/pull/3884

### Summary
Optimized `WeightedAggregationHelper` to use in-place tensor operations, reducing memory usage during aggregation by ~2x for large models. This benefits FedAvg, ScatterAndGather, and all workflows using weighted aggregation.

### Changes
1. **In-place operations**: Use `.mul_()` and `.add_()` for PyTorch tensors instead of creating new tensors
2. **Framework-agnostic**: Duck-typing to support both PyTorch tensors and NumPy arrays without importing torch
3. **Integer tensor handling**: Fixed bug where integer tensors (e.g., batch norm counts) are now summed instead of averaged
4. **Comprehensive tests**: Added 20 unit tests covering PyTorch, NumPy, mixed types, and edge cases

### Memory Impact
- **Before**: `tensor.mul(weight)` creates new tensor → 2x memory temporarily
- **After**: `tensor.mul_(weight)` modifies in-place → 1x memory only

For InTime aggregation, the input tensor is GC'd after the callback, making in-place modification safe.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: tests/unit_test/app_common/aggregators/weighted_aggregation_helper_test.py [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+        helper.add(data1, weight=2.0, contributor_name="site-1", contribution_round=0)
+
+        # Second contribution
+        data2 = {"count": torch.tensor([5, 10, 15], dtype=torch.long)}
+        helper.add(data2, weight=3.0, contributor_name="site-2", contribution_round=0)
+
+        result = helper.get_result()
+
+        expected = torch.tensor([7.0, 14.0, 21.0])
+        assert torch.allclose(result["count"], expected)
```
[the remark this answers is not in the document]
Comment (YuanTingHsieh) [reply tests/unit_test/app_common/aggregators/weighted_aggregation_helper_test.py:71]: right that comment is flawed, logic is correct.

[the remark this answers is not in the document]
Comment (chesterxgchen) [reply tests/unit_test/app_common/aggregators/weighted_aggregation_helper_test.py:71]: update tests I will approve again
